compiler/rustc_interface/src/interface.rs

```diff
@@ -452,6 +448,15 @@ pub fn run_compiler<R: Send>(config: Config, f: impl FnOnce(&Compiler) -> R + Se
                 config.expanded_args,
             );
 
+            // Even though the session holds the lint store, we can't build the
+            // lint store until the session exists. So do that now.
+            let mut lint_store = rustc_lint::new_lint_store(sess.enable_internal_lints());
+            if let Some(register_lints) = config.register_lints.as_deref() {
+                register_lints(&sess, &mut lint_store);
```

Comment:
There are a few initialization routines that change sess after this call. What do callbacks expect to see? Should this initialization happen a bit later in this function?

Reply:
Good suggestion. I've moved the initialization to just before the session is wrapped in an `Lrc` and moved into the `Compiler`.